tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```
@@ -143,6 +143,16 @@ spec:
           local identity=$1
           local reference=$2
           local digest=$3
+
+          # cosign has very limited support for selecting the right auth entry,
+          # so create a custom auth file with just one entry.
+          SOURCE_AUTH_FILE="$(mktemp -d)/config.json"
+
+          select-oci-auth "${reference}" > "$SOURCE_AUTH_FILE"
+
+          DOCKER_CONFIG=$(dirname "${SOURCE_AUTH_FILE}")
+          export DOCKER_CONFIG
```

Sorry I didn't notice this earlier. It seems a bit odd to not save the dir at first and then get it via `dirname`.

```suggestion
          export DOCKER_CONFIG="$(mktemp -d)"
          select-oci-auth "${reference}" > "${DOCKER_CONFIG}/config.json"
```

Cleaner, no? :)

---

looks good. using your suggestion but Checkton was not happy ... needed a small tweak.

---

oh, interesting, I didn't know about that. good to know.

https://github.com/koalaman/shellcheck/wiki/SC2155

looks good.